### ISSUE TITLE
Failing tests in the test suite

### ISSUE THREAD

**Issue by 23Skidoo** [scrive/hpqtypes#4]
There's a number of tests [that are currently failing](
https://travis-ci.org/scrive/hpqtypes/jobs/161871170) in the test suite:

```
pt to get non-NULL value of type JSON Value fails if NULL is provided:
[Failed] ERROR: DBException {dbeQueryContext = SQL "SELECT <Nothing>",
dbeError = DetailedQueryError {qeSeverity = "ERROR", qeErrorCode =
InternalError, qeMessagePrimary = "cache lookup failed for type 114",
qeMessageDetail = Nothing, qeMessageHint = Nothing,
qeStatementPosition = Nothing, qeInternalPosition = Nothing,
qeInternalQuery = Nothing, qeContext = Nothing, qeSourceFile = Just
"tupdesc.c", qeSourceLine = Just 495, qeSourceFunction = Just
"TupleDescInitEntry"}}

pt to get non-NULL value of type JSONB Value fails if NULL is
provided: [Failed] ERROR: DBException {dbeQueryContext = SQL "SELECT
<Nothing>", dbeError = DetailedQueryError {qeSeverity = "ERROR",
qeErrorCode = InternalError, qeMessagePrimary = "cache lookup failed
for type 3802", qeMessageDetail = Nothing, qeMessageHint = Nothing,
qeStatementPosition = Nothing, qeInternalPosition = Nothing,
qeInternalQuery = Nothing, qeContext = Nothing, qeSourceFile = Just
"tupdesc.c", qeSourceLine = Just 495, qeSourceFunction = Just
"TupleDescInitEntry"}}

[...]
```

and the like. IIUC, those cache lookup errors are caused by race conditions. Not clear yet whether the bug is in the library itself or only the test suite.

**Comment by arybczak** [comment]
More likely the PostgreSQL version available there doesn't support JSON/JSONB.

**Comment by 23Skidoo** [comment]
@arybczak Thanks, looks like you're right, it used 9.1. Switched to 9.4.

**Comment by 23Skidoo** [comment]
Yay, that fixed it, see 6ff504b4ec9a5c01067f06d17457fd053f56c5d8. See [here](http://stackoverflow.com/a/32899097/242169) for the explanation why the first attempt (96aadf2f5156f81fa6ecc96092e3c8ae70d0dbe5) didn't work.
